public/src/main/java/com/revenuecat/purchases/utils/Iso8601UtilsDupe.kt

```diff
@@ -0,0 +1,251 @@
+package com.revenuecat.purchases.utils
+
+import java.util.Calendar
+import java.util.Date
+import java.util.GregorianCalendar
+import java.util.TimeZone
+
+/**
+ * TODO this is a dupe of ISO8601Utils.java in common module, but it's used in the public module so we can't remove it yet,
+ * nor make it public here because it would be accessible to everyone
+ */
+/*
+ * Copyright (C) 2011 FasterXML, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Jackson’s date formatter, pruned to Moshi's needs. Forked from this file:
+ * https://github.com/FasterXML/jackson-databind/blob/master/src/main/java/com/fasterxml/jackson/databind/util/ISO8601Utils.java
+ *
+ *
+ * Utilities methods for manipulating dates in iso8601 format. This is much much faster and GC
+ * friendly than using SimpleDateFormat so highly suitable if you (un)serialize lots of date
+ * objects.
+ *
+ *
+ * Supported parse format:
+ * [yyyy-MM-dd|yyyyMMdd][T(hh:mm[:ss[.sss]]|hhmm[ss[.sss]])]?[Z|[+-]hh[:]mm]]
+ *
+ * @see [this specification](http://www.w3.org/TR/NOTE-datetime)
+ */
+internal object Iso8601UtilsDupe {
+    /** ID to represent the 'GMT' string  */
+    private const val GMT_ID = "GMT"
+
+    /** The GMT timezone, prefetched to avoid more lookups.  */
+    private val TIMEZONE_Z = TimeZone.getTimeZone(GMT_ID)
+
+    /**
+     * Parse a date from ISO-8601 formatted string. It expects a format
+     * [yyyy-MM-dd|yyyyMMdd][T(hh:mm[:ss[.sss]]|hhmm[ss[.sss]])]?[Z|[+-]hh:mm]]
+     *
+     * @param date ISO string to parse in the appropriate format.
+     * @return the parsed date
+     */
+    fun parse(date: String): Date {
+        return try {
+            var offset = 0
+
+            // extract year
+            val year = parseInt(date, offset, 4.let { offset += it; offset })
```

Comment:
This is probably not the prettiest code... I've been thinking about simply rewriting this (note that once we join the modules we still need to make this internal), but maybe we can do that in a separate PR